File: agents/java-agent/src/main/java/com/hmdev/messaging/agents/api/ConnectionChannelApi.java
package com.hmdev.messaging.agents.api;

import com.hmdev.messaging.agents.utils.ApiResponse;

public interface ConnectionChannelApi {

	public ApiResponse connect(String channelName,String channelKey,String user) throws Exception;	
	public ApiResponse receive(String channelKey,String session,String range);	
	public ApiResponse send(String msg,String fromUsr,String toUser,String key,String session);
	public ApiResponse disconnect(String channelKey,String session);	

}

File: agents/python-agent/hmdev/messaging/agent/api/connection_channel_api.py
from abc import ABC, abstractmethod
from typing import List, Optional

from hmdev.messaging.agent.api.models import ConnectResponse, EventMessageResult, AgentInfo, ReceiveConfig


class ConnectionChannelApi(ABC):
    @abstractmethod
    def connect(self, channel_name: str, channel_key: str, agent_name: str, session_id: str | None = None) -> ConnectResponse:
        ...

    @abstractmethod
    def receive(self, session_id: str, receive_config: ReceiveConfig) -> Optional[EventMessageResult]:
        ...

    @abstractmethod
    def get_active_agents(self, session_id: str) -> Optional[List[AgentInfo]]:
        ...

    @abstractmethod
    def send(self, msg: str, to_user: str | None, session_id: str) -> bool:
        ...

    @abstractmethod
    def disconnect(self, session_id: str) -> bool:
        ...

    # UDP bridge operations via the messaging-service UDP listener
    @abstractmethod
    def udp_push(self, msg: str, to_user: str | None, session_id: str) -> bool:
        ...

    @abstractmethod
    def udp_pull(self, session_id: str, receive_config: ReceiveConfig) -> Optional[EventMessageResult]:
        ...
